pkgs/build-support/rust/import-cargo-lock.nix

```
@@ -117,9 +119,13 @@ let
       gitParts = parseGit pkg.source;
       registryIndexUrl = lib.removePrefix "registry+" pkg.source;
     in
-      if lib.hasPrefix "registry+" pkg.source && builtins.hasAttr registryIndexUrl registries then
+      if ( lib.hasPrefix "registry+" pkg.source && builtins.hasAttr registryIndexUrl registries )
+        || ( lib.hasPrefix "sparse+" pkg.source && builtins.hasAttr pkg.source registries ) then
```

I'm not sure about the formatting of the parentheses here, maybe it should be this? Not really too big of a deal though, as `if (` hasn't been used anywhere in this file before. Will let others weigh in.

```suggestion
      if (lib.hasPrefix "registry+" pkg.source && builtins.hasAttr registryIndexUrl registries)
        || (lib.hasPrefix "sparse+" pkg.source && builtins.hasAttr pkg.source registries) then
```

---

If in other files the common use is without spaces makes more sense, so it's modified.